### ISSUE TITLE
readFloatLengthCoded does not return expected DOUBLE value.

### ISSUE THREAD

**Issue by RichardMcSorley** [mariadb-corporation/mariadb-connector-nodejs#119]
In the case where a double value is set to `-0.9999237060546875` in the Database, [This function ](https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/709ef9476ba777cc918b1641ffe235222749dc2b/lib/io/packet.js#L384
) returns `-0.9999237060546876`

Expected `-0.9999237060546875` but received `-0.9999237060546876`

While stepping over the code I can see the values are as such:

```javascript
end = this.pos + len // = 221
```

The code steps into [this for loop](https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/709ef9476ba777cc918b1641ffe235222749dc2b/lib/io/packet.js#L409) and the last 2 iterations are as follows:

```javascript
this.pos //220
this.buf[this.pos ++] // = 17
String.fromCharCode(this.buf[this.pos ++]) // = "" charCode
factor * (result + resultDot / dotfactor) // = -0.999923706054687      
```

```javascript
this.pos // = 221
    this.buf[this.pos ++] // = 54
String.fromCharCode(this.buf[this.pos ++]) // = "6" charCode
    factor * (result + resultDot / dotfactor) //  = -0.9999237060546876       
```

Some environment info:

```bash
$ node -v
v12.18.1
```

Library version: `mariadb: ^2.3.1`

**Comment by rusher** [comment]
Can you add debug to your configuration, like this : 
`const con =mariadb.createConnection({host: 'mydb.com', user: 'myUser', password: 'myPwd', debug: true});`

Code here is like atof (better than using parseFloat, because parseFloat does many other unusefull things for our use case), so it seems it represent what server has sent (ascii 54 = 6).

**Comment by RichardMcSorley** [comment]
@rusher I set debug to `true` and I see the log outputs:

```bash
 <== conn:1555295 Query.readResultSetRow (0,20)
+--------------------------------------------------+
|  0  1  2  3  4  5  6  7   8  9  a  b  c  d  e  f |
+--------------------------------------------------+------------------+
| 14 00 00 12 13 2D 30 2E  39 39 39 39 32 33 37 30 | .....-0.99992370 |
| 36 30 35 34 36 38 37 35                          | 60546875         |
+--------------------------------------------------+------------------+
```

**Comment by RichardMcSorley** [comment]
Here is another value where this happens:
```bash
<== conn:1519704 Query.readResultSetRow (0,20)
+--------------------------------------------------+
|  0  1  2  3  4  5  6  7   8  9  a  b  c  d  e  f |
+--------------------------------------------------+------------------+
| 14 00 00 04 13 2D 30 2E  39 39 39 36 37 39 35 36 | .....-0.99967956 |
| 35 34 32 39 36 38 37 35                          | 54296875         |
+--------------------------------------------------+------------------+
```
Expected `0.9999237060546875` but received `-0.9996795654296876`

**Comment by RichardMcSorley** [comment]
If the last digits are lesser or higher than 5 they are return the expected value.

```
-0.9999237060546874 => -0.9999237060546874 
-0.9999237060546876 => -0.9999237060546876
```

**Comment by RichardMcSorley** [comment]
Updated the title from ~~FLOAT~~ to DOUBLE, the value is stored as a DOUBLE in our table

**Comment by RichardMcSorley** [comment]
@rusher Confirmed that the Java Library https://github.com/mariadb-corporation/mariadb-connector-j is returning the expected value but this javascript library seems to still have this issue.

**Comment by rusher** [comment]
how. That so obvious, the atof function will have some problem when integer or fractionnal part is > than 9007199254740991  due to int limitation (ieee-754).
I'll deal with that.

**Comment by rusher** [comment]
corrected with commit https://github.com/mariadb-corporation/mariadb-connector-nodejs/commit/2bfe64fa516850347f464bf082951b8edab15af4

**Comment by rusher** [comment]
released in 2.4.2 version

**Comment by RichardMcSorley** [comment]
Thanks so much @rusher! 😃👍
